### ISSUE TITLE
RFC:  mark non-redistributable materials

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#921]
Soon we will approach git annex meta-data use and I wondered if we should agree on some meta-data tags/fields to use, especially to signal which content should not be published by 'publish'

- **proprietary** content, e.g. stimuli movie files
- human subject **sensitive** data, e.g. not defaced anatomicals
- data files which are generally non-redistributable since original source requires **authentication**

So I wondered if we should may be use those bold terms as annex metadata tags, and then by default not publish anything which has those tags... ?  

actually -- since we might grow the number of such 'restrictions', what about making a metadata field "redistribution-restrictions" which would collect all such restrictions, so we could easily say to not publish anything which has any `redistribution-restrictions`, e.g. (here it is to match, but we could say `--not`; also here named `restrictions` but we could make it more descriptive with `redistribution-restrictions`)

```shell
$> git annex find --not --in here --metadata restrictions=\*         
stimulus/task002/orig/INDIANA_JONES_RAIDERS_LOST_ARK_part_1.m4v
```

what do you think @datalad/developers ?

**Comment by mih** [comment]
I am not sure whether we need to classify different reasons for not publishing pieces. We could simply have **donotpublish**. If we want to classify, I'd prefer to have general labels like **sensitive**, rather than *non-defaced*.

**Comment by yarikoptic** [comment]
we don't need to but I think it would be better to classify so we ourselves (and others) know why specific file shouldn't be redistributed without consideration.  May be with some collaborators I would like to share one type of data (e.g. "sensitive") but not the other (e.g. "proprietary"), which then could be easily done if we separate those out.  And it is still as easy to say "with any restriction"

**Comment by mih** [comment]
True.

**Comment by bpoldrack** [comment]
Do you see a way to let the users specify the classification terms while `datalad` is just providing the mechanism to allow for such definitions? Depending on the field the data is about it might heavily differ what terms really make sense. And we don't want to have it convenient for neuroscientists, while everybody else has to use **sensitive** for everything, do we?
Otherwise I agree.

**Comment by bpoldrack** [comment]
Actually, we can use any tag to specify `redistribution-restrictions`, right? So we probably need just a proxy command to assign tags (recursively).

**Comment by yarikoptic** [comment]
Yes, we need helpers for metadata. Be careful -  Annex metadata has tags (entries without values) and actual key=value entries. And those are somewhat different I think ;-)

**Comment by yarikoptic** [comment]
a sample case was http://datasets.datalad.org/?dir=/labs/haxby/raiders and it worked out nicely

**Comment by mih** [comment]
Looking into this again, I think we should not diversify tags with the same intended consequences too much. In order to have publish not publish content with restrictions we need to set up default preferred content expressions. Those are per local remote, hence cannot be informed by the original dataset creator. The only way I see this work is to have the creator use a defined meta data tag for labeling restricted content, and have `create-sibling` always set up a preferred content expression that prevents such data from going to a remote, unless explicitly instructed that it is OK to accept restricted content.

- [ ] The `crawler` needs to start marking all file content that is from behind an auth-wall with a tag (maybe `auth-required`).
- [ ] `create-sibling` needs to get a switch to override publication restrictions (which need to be ON be default, in order to avoid accidental publication), for example in the context of a "backup" remote (#925)

**Comment by yarikoptic** [comment]
1. agreed that crawler should mark all restricted content!  I was just waiting for us to reach conclusion to define precisely names and values for the meta-data to be used. So far I think I have used only `distribution-restrictions=sensitive` (and `distribution-restrictions=proprietary`), so if no objections, could add `auto-required`, so it could then look like (just an example on raiders ds):
```shell
hopa:/tmp/raiders/sub001/anatomy
$> git annex metadata --set distribution-restrictions+=auth-required highres004.PAR
metadata highres004.PAR 
  dcterms_format=image/parrec
  dcterms_format-lastchanged=2016-09-26@19-37-46
  distribution-restrictions=auth-required
  distribution-restrictions=sensitive
  distribution-restrictions-lastchanged=2017-02-14@15-59-46
  lastchanged=2017-02-14@15-59-46
ok
(recording state in git...)
```

2. `create-sibling` indeed should be capable to specify pretty much either some standard annex group (https://git-annex.branchable.com/preferred_content/standard_groups/) or some custom ones we add (e.g. to not publish anything with restrictions).  But I think that by default it should either do nothing, or set it to `manual` group.  Then `publish` by default would copy no data, and would copy to remote host data only if either explicit PATH is provided, or no PATHs provided but some annex wanted setting/group is specified so `annex copy --auto` does its job (see https://github.com/datalad/datalad/issues/1274#issuecomment-279747603 for more detail/analysis)

**Comment by bpoldrack** [comment]
Re 2: Not sure, whether I fully understand your wishes for `publish`, @yarikoptic .
Do you want `publish some/path` to ignore those restrictions without any switch explicitly saying to do so?

**Comment by yarikoptic** [comment]
> Do you want `publish some/path` to ignore those restrictions without any switch explicitly saying to do so?

that is a good question.  Current behavior -- yes, AFAIK we effectively ignore those `wanted` settings whenever specific paths are provided to `publish` since they are in effect only for `annex copy --auto` which we invoke **only** when no paths are provided.  We might want to reconsider and see how to fix that.
